enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -61,9 +62,11 @@
 
   Writes (even nested) yaml keys in a deterministic fashion."
   [filename obj]
-  (io/make-parents filename)
-  (spit filename (yaml/generate-string (serialization-deep-sort obj)
-                                       {:dumper-options {:flow-style :block :split-lines false}})))
+  (if (u.files/make-writable-parents! filename)
```

Comment:
Rather than make a new untested util function with at least one weird edge case, I think it would be safer to just inline the check in this function body, like this
```
  (when (not (.canWrite (.getParentFile (io/file filename))))
    (throw (ex-info (format "Destination path is not writeable: %s" filename) {:filename filename})))
```

Reply:
sure, why not